base_comment_template/models/base_comment_template.py

```diff
@@ -124,37 +101,29 @@ class BaseCommentTemplate(models.Model):
         help="This comment template will be available on this models. "
         "You can see here only models allowed to set the coment template.",
     )
-
     domain = fields.Char(
-        "Filter Domain",
+        string="Filter Domain",
         required=True,
         default="[]",
         help="This comment template will be available only for objects "
         "that satisfy the condition",
     )
-
-    priority = fields.Integer(
-        default=10, copy=False, help="the highest priority = the smallest number",
+    sequence = fields.Integer(
+        required=True, default=1, help="The smaller number = The higher priority"
     )
 
-    @api.constrains("domain", "priority", "model_ids", "position")
-    def _check_partners_in_company_id(self):
-        templates = self.search([])
-        for record in self:
-            other_template_same_models_and_priority = templates.filtered(
-                lambda t: t.priority == record.priority
-                and set(record.model_ids).intersection(record.model_ids)
-                and t.domain == record.domain
-                and t.position == record.position
-                and t.id != record.id
+    def name_get(self):
+        """Redefine the name_get method to show the template name with the position.
+        """
+        res = []
+        for item in self:
+            name = "{} ({})".format(
+                item.name, dict(self._fields["position"].selection).get(item.position)
             )
-            if other_template_same_models_and_priority:
-                raise ValidationError(
-                    _(
-                        "There are other records with same models, priority, "
-                        "domain and position."
-                    )
-                )
+            if self.env.context.get("add_models_in_name"):
+                name = "{} Apply on: {}".format(name, item.model_ids.mapped("name"))
```

Comment:
This is not translatable, and not correctly spelled (it would be `Applied on:`), but let's keep it minimal:
```suggestion
                name += " (%s)" % ", ".join(item.model_ids.mapped("name"))
```